projects/js-packages/components/components/text/index.tsx

```diff
@@ -0,0 +1,99 @@
+/**
+ * External dependencies
+ */
+import React, { useMemo } from 'react';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import styles from './style.module.scss';
+import type { H3Props, TextProps, TitleProps } from './types';
+import { BOX_MODEL_VALUES, VARIANTS_MAPPING } from './constants';
+
+/**
+ * Text component.
+ *
+ * @param {TextProps} props - Component props.
+ * @returns {React.ReactElement} - JSX.Element
+ */
+const Text: React.FC< TextProps > = ( {
+	variant = 'body',
+	children,
+	component,
+	className,
+	...componentProps
+} ) => {
+	const Component = ( component || VARIANTS_MAPPING[ variant ] || 'span' ) as React.FC< TextProps >;
```

Comment:
Can we export the `component` piece at `types` and reuse it here?

Reply:
It should work, but if you get the error message, it's resolving to `@types/react@18` for some reason:

![image](https://user-images.githubusercontent.com/1663717/166969362-c1c08b66-325c-44a9-8f3c-3e52d93a1c91.png)

Reply:
Yes, there is some version mismatch that makes it worse. We can fix it after we upgrade to React 18

Reply:
Done, thanks

Reply:
@manzoorwanijk btw, I think we could bump `@types/react` to 18, don't get any problem doing it here.

Reply:
Sounds good.

Reply:
Done and we. no longer need that casting 🎉
Thank you.